fedcloud_monitoring_tools/vo_mon.py

```diff
@@ -0,0 +1,157 @@
+"""VO-level testing for a given site in a VO"""
+
+import io
+import os
+import time
+from datetime import datetime, timezone
+
+import click
+import paramiko
+from fabric import Connection
+from imclient import IMClient
+
+IM_REST_API = "https://im.egi.eu/im"
+AUTH_FILE = "auth.dat"
+MAX_NUM_CHECKS = 10
+
+
+class VOTestException(Exception):
+    pass
+
+
+class VOTest:
+    """Helper class to call im-client easily"""
+
+    def __init__(self, vo, site, token):
+        self.vo = vo
+        self.site = site
+        self.token = token
+        self.now = datetime.now(timezone.utc)
+
+    def create_auth_file(self, filepath):
+        with open(filepath, "w") as output_file:
+            output_file.write(
+                "id = im; type = InfrastructureManager; token = {}\n".format(self.token)
+            )
+            output_file.write(
+                "id = egi; type = EGI; host = {}; vo = {}; token = {}\n".format(
+                    self.site, self.vo, self.token
+                )
+            )
+
+    def delete_auth_file(self, filepath):
```

Comment:
See comment above about using tempfile, so you do not need to care about deleting